cryptofeed/pairs.py

```diff
@@ -250,62 +265,78 @@ def okcoin_pairs():
+        # swaps
+        r = requests.get('https://www.okex.com/api/swap/v3/instruments/ticker').json()
+        for update in r:
+            data[update['instrument_id']] = update['instrument_id']
+        # futures
+        r = requests.get('https://www.okex.com/api/futures/v3/instruments/ticker').json()
+        for update in r:
+            data[update['instrument_id']] = update['instrument_id']
+        return data
+    except Exception as why:
+        LOG.exception("OKEX unexpected response: %r %r", r, why)
+        raise
+
+
+def coinbene_pairs() -> Dict[str, str]:
+    response = requests.get('http://api.coinbene.com/v1/market/symbol')
+    try:
+        r = response.json()
+        return {f"{e['baseAsset']}{PAIR_SEP}{e['quoteAsset']}": e['ticker'] for e in r['symbol']}
+    except Exception as why:
+        LOG.critical("COINBENE: Skip because %r (status=%r) in unexpected JSON: %.500r", why, response.status_code, response.text)
+        raise
+
+
+def bittrex_pairs() -> Dict[str, str]:
     r = requests.get('https://api.bittrex.com/api/v1.1/public/getmarkets').json()
     r = r['result']
     return {f"{e['MarketCurrency']}{PAIR_SEP}{e['BaseCurrency']}": e['MarketName'] for e in r if e['IsActive']}
 
 
-def bitcoincom_pairs():
+def bitcoincom_pairs() -> Dict[str, str]:
     r = requests.get('https://api.exchange.bitcoin.com/api/2/public/symbol').json()
     return {f"{data['baseCurrency']}{PAIR_SEP}{data['quoteCurrency'].replace('USD', 'USDT')}": data['id'] for data in r}
 
 
-def bitmax_pairs():
-    r = requests.get('https://bitmax.io/api/v1/products').json()
-    return {f"{data['baseAsset']}{PAIR_SEP}{data['quoteAsset']}": data['symbol'] for data in r}
+def bitmax_pairs() -> Dict[str, str]:
+    r = None
+    try:
+        r = requests.get('https://bitmax.io/api/v1/products')
+        return {f"{data['baseAsset']}{PAIR_SEP}{data['quoteAsset']}": data['symbol'] for data in r.json()}
+    except Exception as why:
+        LOG.error("BITMAX Unexpected message: %r %r", r.text, why)
+        raise
 
 
-def upbit_pairs():
+def upbit_pairs() -> Dict[str, str]:
     r = requests.get('https://api.upbit.com/v1/market/all').json()
     return {f"{data['market'].split('-')[1]}{PAIR_SEP}{data['market'].split('-')[0]}": data['market'] for data in r}
 
 
-def blockchain_pairs():
+def blockchain_pairs() -> Dict[str, str]:
     r = requests.get("https://api.blockchain.com/mercury-gateway/v1/instruments").json()
     return {data["symbol"].replace("-", PAIR_SEP): data["symbol"] for data in r}
 
 
-def gateio_pairs():
+def gateio_pairs() -> Dict[str, str]:
     r = requests.get("https://api.gateio.ws/api/v4/spot/currency_pairs").json()
     return {data['id'].replace("_", PAIR_SEP): data['id'] for data in r}
 
 
-def bitmex_pairs():
+def bitmex_pairs() -> Dict[str, str]:
     r = requests.get("https://www.bitmex.com/api/v1/instrument/active").json()
     return {entry['symbol']: entry['symbol'] for entry in r}
```

Comment:
hi @olibre 
I am sorry, I don't understand. Nothing has changed here regarding Bitmex pair normalization. You indicate that this PR implements pair normalization for Bitmex. Am I being dumb?

Reply:
Yes @yohplala you are right. I am confused => I have fixed this PR title. I will fix the Bitmex pair retrieval later...

Reply:
Yes @yohplala you are right. I am confused => I have fixed this PR title.